src/codec_aom.c

```diff
@@ -606,26 +607,30 @@ static avifResult aomCodecEncodeImage(avifCodec * codec,
                 avifDiagnosticsPrintf(codec->diag, "aom_codec_enc_config_default() failed: %s", aom_codec_err_to_string(err));
                 return AVIF_RESULT_UNKNOWN_ERROR;
             }
+        } else {
+            cfg->g_usage = aomUsage;
```

Comment:
Done. `encoder->speed` decides `aomUsage` and `aomCpuUsed`, so updating `g_usage` here. In "init" code path this field is set in `aom_codec_enc_config_default()`.

Reply:
I see. Your message helped me more than the comment though, so maybe consider using the former as the latter. Something like `// aomUsage was taken into account by aom_codec_enc_config_default() but it can only be manually updated afterwards.`

Reply:
Changing `cfg->g_usage` like this is risky, because some other fields in `cfg` may be tightly coupled to `cfg->g_usage` and need to be changed accordinly. It is best to avoid this potential issue, by not allowing `encoder->speed` to change.

Similarly, it is best to not allow `encoder->maxThreads` to change.

Reply:
I looked into this. I found that libaom's test/datarate_test.h changes `AOME_SET_CPUUSED` dynamically in the middle of a video sequence (search for `speed_change_test_` in the file), and I also saw this done in webrtc's libaom_av1_encoder.cc file. So I believe libaom supports this. However, libavif may need to change `cfg->g_usage` at the same time it changes `AOME_SET_CPUUSED`, so it is best to avoid this until it's really needed.